bigtable/client/testing/embedded_server_test_fixture.h

```diff
@@ -0,0 +1,128 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_TESTING_EMBEDDED_SERVER_TEST_FIXTURE_H_
+#define GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_TESTING_EMBEDDED_SERVER_TEST_FIXTURE_H_
+
+#include "bigtable/client/table.h"
+#include "bigtable/client/table_admin.h"
+#include "bigtable/client/testing/inprocess_admin_client.h"
+#include "bigtable/client/testing/inprocess_data_client.h"
+#include "google/bigtable/v2/bigtable.grpc.pb.h"
+#include <gtest/gtest.h>
+
+namespace bigtable {
+namespace testing {
+
+using ReceivedMetadata = std::multimap<std::string, std::string>;
+
+inline void GetClientMetadata(grpc::ServerContext* context,
+                              ReceivedMetadata& client_metadata) {
+  for (auto it = context->client_metadata().begin();
+       it != context->client_metadata().end(); ++it) {
+    auto ele =
+        std::make_pair(std::string(it->first.begin(), it->first.end()),
+                       std::string(it->second.begin(), it->second.end()));
+    client_metadata.emplace(ele);
```

Comment:
If you want the `emplace()` to use move semantics it should be `client_metadata.emplace(std::move(ele))` right?

Reply:
done that